tests/test_high_level_io.py

```diff
@@ -56,6 +57,81 @@ def ang_lightcone(ic, lc):
     )
 
 
+def test_read_bad_file_lc(test_direc, lc):
+    # create a bad hdf5 file with some good fields,
+    #  some bad fields, and some missing fields
+    #  in both input parameters and box structures
+    fname = lc.save(direc=test_direc)
+    with h5py.File(fname, "r+") as f:
+        # make gluts, these should be ignored on reading
+        f["user_params"].attrs["NotARealParameter"] = "fake_param"
+        f["_globals"].attrs["NotARealGlobal"] = "fake_param"
+
+        # make gaps
+        del f["user_params"].attrs["BOX_LEN"]
+        del f["_globals"].attrs["OPTIMIZE_MIN_MASS"]
+
+    # load without compatibility mode, make sure we throw the right error
+    with pytest.raises(ValueError, match="There are extra or missing"):
+        LightCone.read(fname, direc=test_direc, safe=True)
+
+    # load in compatibility mode, check that we warn correctly
+    with pytest.warns(UserWarning, match="There are extra or missing"):
+        lc2 = LightCone.read(fname, direc=test_direc, safe=False)
+
+    # check that the fake fields didn't show up in the struct
+    assert not hasattr(lc2.user_params, "NotARealParameter")
+    assert "NotARealGlobal" not in lc2.global_params.keys()
+
+    # check that missing fields are set to default
+    assert lc2.user_params.BOX_LEN == UserParams._defaults_["BOX_LEN"]
+    assert lc2.global_params["OPTIMIZE_MIN_MASS"] == global_params.OPTIMIZE_MIN_MASS
```

Comment:
This assert would be more useful if I changed the global_params in the lc/coeval fixtures